### PR TITLE
feat: [CDS-70307]: GitOps changes for Org level services and clusters mapping

### PR DESCRIPTION
**Pull request by meenaravichandran1**
## Describe your changes
Add org level services and clusters mapping (for linked environments) changes

## Checklist
- [ ] I've documented the changes in the PR description.
- [ ] I've tested this change either in PR or local environment.
- [ ] If hashcheck failed I followed [the checklist](https://harness.atlassian.net/wiki/spaces/DEL/pages/21016838831/PR+Codebasehash+Check+merge+checklist) and updated this PR description with my answers to make sure I'm not introducing any breaking changes.

## Comment Triggers
<details>
  <summary>Build triggers</summary>

- Feature build: `trigger feature-build`
- 
  Specific builds
  - `trigger manager`
  - `trigger dms`
  - `trigger ng_manager`
  - `trigger cvng `
  - `trigger cmdlib`
  - `trigger template_svc`
  - `trigger events_fmwrk_monitor`
  - `trigger event_server`
  - `trigger change_data_capture`
  -  Trigger multiple builds together: For eg: `trigger dms, manager`
- Immutable delegate `trigger publish-delegate`
</details>

## [Latest PR Check Triggers](https://github.com/harness/harness-core/blob/develop/.github/pull_request_template.md)

<details>
  <summary>PR Check triggers</summary>
You can run multiple PR check triggers by comma separating them in a single comment. e.g. `trigger ut0, ut1`

- Compile: `trigger compile`
- CodeformatCheckstyle: `trigger checkstylecodeformat`
  - CodeFormat: `trigger codeformat`
  - Checkstyle: `trigger checkstyle`
- MessageMetadata: `trigger messagecheck`
- File-Permission-Check: `trigger checkpermission`
- Recency: `trigger recency`
- BuildNumberMetadata: `trigger buildnum`
- Trigger CommonChecks: `trigger commonchecks`
- PMD: `trigger pmd`
- Copyright Check: `trigger copyrightcheck`
- Feature Name Check: `trigger featurenamecheck`
- UnitTests-ALL: `trigger uts`
- UnitTests-0: `trigger ut0`
- UnitTests-1: `trigger ut1`
- UnitTests-2: `trigger ut2`
- UnitTests-3: `trigger ut3`
- UnitTests-4: `trigger ut4`
- UnitTests-5: `trigger ut5`
- UnitTests-6: `trigger ut6`
- UnitTests-7: `trigger ut7`
- UnitTests-8: `trigger ut8`
- UnitTests-9: `trigger ut9`
- CodeBaseHash: `trigger codebasehash`
- CodeFormatCheckstyle: `trigger checkstylecodeformat`
- SonarScan: `trigger ss`
- GitLeaks: `trigger gitleaks`
- Trigger all Checks: `trigger smartchecks`
- Go Build: `trigger gobuild`
- Validate_Reviews: `trigger review`
- Module Dependency Check: `trigger mdc`
</details>

## PR check failures and solutions
https://harness.atlassian.net/wiki/spaces/BT/pages/21106884744/PR+Checks+-+Failures+and+Solutions

## [Contributor license agreement](https://github.com/harness/harness-core/blob/develop/CONTRIBUTOR_LICENSE_AGREEMENT.md)

- - -
This change is [Reviewable](https://reviewable.io/reviews/harness/harness-core/49504)

### CODE REVIEW

**Comment by meenaravichandran1** [comment]
trigger uts

File: 120-ng-manager/src/main/java/io/harness/ng/gitops/resource/ClusterResource.java [hunk 1]
```diff
@@ -333,19 +333,29 @@ public class ClusterResource {
...
             .flatMap(List::stream)
-            .collect(Collectors.toMap(e
-                -> e.getScopeLevel().toString().toLowerCase() + "." + e.getIdentifier(),
-                Function.identity(), (c1, c2) -> c1));
+            .collect(Collectors.toMap(
+                e -> getScope(e.getScopeLevel()) + "." + e.getIdentifier(), Function.identity(), (c1, c2) -> c1));
     return ResponseDTO.newResponse(getNGPageResponse(entities.map(e -> ClusterEntityMapper.writeDTO(e, allClusters))));
   }
 
+  private String getScope(ScopeLevel scopeLevel) {
```
Comment (manavjot-harness) [inline 120-ng-manager/src/main/java/io/harness/ng/gitops/resource/ClusterResource.java:352]: nit: Please add a comment on why we needed this method for orgs only.

Comment (meenaravichandran1) [reply 120-ng-manager/src/main/java/io/harness/ng/gitops/resource/ClusterResource.java:352]: done

File: 120-ng-manager/src/main/java/io/harness/ng/gitops/resource/ClusterResource.java [hunk 2]
```diff
@@ -333,19 +333,31 @@ public class ClusterResource {
...
+    // For Organization scoped clusters, the prefix used is "org"
+    if (ScopeLevel.ORGANIZATION.equals(scopeLevel)) {
+      return "org";
```
Comment (lasdox) [inline 120-ng-manager/src/main/java/io/harness/ng/gitops/resource/ClusterResource.java:355]: I suggest you to create a constant like GITOPS_ORGANIZATION_PREFIX or a enum method like "getGitopsPrefix" for this.
This is a difficult change to debug

File: 127-cd-nextgen-entities/src/main/java/io/harness/cdng/gitops/mappers/ClusterEntityMapper.java [hunk 3]
```diff
@@ -158,7 +164,7 @@ public ScopeAndRef getScopeFromClusterRef(String ref) {
       return new ScopeAndRef(ref, split[1], ScopeLevel.ACCOUNT);
     }
 
-    if (ScopeLevel.ORGANIZATION.toString().equalsIgnoreCase(split[0])) {
+    if (ORG.equalsIgnoreCase(split[0])) {
```
Comment (lasdox) [inline 127-cd-nextgen-entities/src/main/java/io/harness/cdng/gitops/mappers/ClusterEntityMapper.java:167]: IMO we should have one centralized const or enum method

**Comment by meenaravichandran1** [comment]
trigger uts

**Review by manavjot-harness — APPROVED** [review]

File: 120-ng-manager/src/main/java/io/harness/ng/gitops/resource/ClusterResource.java (hunk 2 above)
Comment (meenaravichandran1) [reply 120-ng-manager/src/main/java/io/harness/ng/gitops/resource/ClusterResource.java:355]: will make this in the next PR

Comment (meenaravichandran1) [reply 120-ng-manager/src/main/java/io/harness/ng/gitops/resource/ClusterResource.java:355]: adding as enum method will need some changes for the other flows as well (account, project) needing changes to the existing logic, hence created a constant

**Pull request merged**